version: "3.9"
services:
  # Front end
  s2f-front:
    build: 
      context: ./frontend
      dockerfile: Dockerfile.prod
    command: gunicorn frontend.wsgi:application --bind 0.0.0.0:80 -w 4
    volumes:
      - ./frontend/static:/home/app/frontend/staticfiles
      - ./frontend/uploads:/home/app/frontend/uploads
      - ./frontend/jinja2:/home/app/frontend/jinja2
      - ./frontend/base:/home/app/frontend/base
      - ./frontend/frontend:/home/app/frontend/frontend
      - ./data:/home/app/data
      - ./aws-credentials.secret:/home/app/.aws/credentials:ro
    # uncomment these to access the app directly for test purposes, 
    # but expect everything to be broken.
    env_file: 
      - ./env.frontend.secret
    networks:
      - s2f
    depends_on:
      s2f-front-db:
        condition: service_healthy
  s2f-front-db:
    image: postgres:15.6-alpine
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 30s
      timeout: 5s
      retries: 10 # this might seem too many retries, but there is data being copied...
    volumes:
      - s2f-front-db_data:/var/lib/postgresql/data/
    env_file:
      - ./env.s2f-front-db.secret
    networks:
      - s2f
  # Proxy server (used locally only)
  # proxy:
  #   build: ./reverse-proxy
  #   volumes:
  #     - ./reverse-proxy/default.conf:/etc/nginx/conf.d/nginx.conf
  #     - ./reverse-proxy/certs:/etc/nginx/ssl
  #     - ./frontend/static:/home/app/staticfiles
  #   networks:
  #     - s2f
  #   ports:
  #     - 80:80
  #     - 443:443
  #   depends_on:
  #     - s2f-front
  # Back end
  # s2f-back:
  #   build: 
  #     context: ./polling-backend
  #     dockerfile: Dockerfile.prod
  #   command: python client_main.py
  #   env_file: 
  #     - ./env.backend.secret
  #   networks:
  #     - s2f
  #   depends_on:
  #     s2f-back-db:
  #       condition: service_healthy
  # s2f-back-db:
  #   image: postgres:16.6-alpine
  #   healthcheck:
  #     test: ["CMD-SHELL", "pg_isready"]
  #     interval: 30s
  #     timeout: 5s
  #     retries: 10 # this might seem too many retries, but there is data being copied...
  #   volumes:
  #     - s2f-back-db_data:/var/lib/postgresql/data/
  #   env_file:
  #     - ./env.s2f-back-db.secret
  #   networks:
  #     - s2f
networks:
  s2f:
    name: s2f
    external: false

volumes:
  s2f-front-db_data:
  # s2f-back-db_data:
  static_s2f:
